2-RunningWithScissors first complete relese

PrzemekSwiderski opened the pull request:
Wydanie działającej gry. Zaimplementowano wszystkie zaplanowane mechaniki.
Brak edycji wizualnej (kolorów, rozbudowanych instrukcji i tekstów) z powodu braku czasu.

Popieprzyłem wysyłanie comitów :(

andramil committed:
Running With Scissors
Napisanie encji

aPusio committed:
learning

andramil committed:
Running With Scissors
Stworzenie kompletnych Encji wraz z DAO obsługującym je

andramil committed:
Running With Scissors
Drobne poprawki graficzne

PrzemekSwiderski committed:
Running with scisors - stworzenie kompletne pakietu gracza, napisanie mechaniki nowej gry oraz wczytania gry.

andramil committed:
Merge branch '9-sos-game' into RWS-Game

andramil committed:
Running With Scissors
Zaimplementowanie interfejsu gry App
ukończenie całego GameService
poprawki w Dao i Service

andramil committed:
Running With Scissors
Działająca gra,
Brak testów jednostkowych.

PrzemekSwiderski committed:
Delete AppTest.java

Nie powinienem tego wysłać

andramil committed:
Revert "Delete AppTest.java"

This reverts commit 332a48aa4fab24a2b228789ff9a30b7cae668528.

aPusio commented on src/main/java/sda/training/App.java:
```diff
@@ -1,55 +1,40 @@
+            int gameNumber = scanner.nextInt();
+            switch (gameNumber) {
+                case 1:
+                    sda.training.rps.app.App app = new sda.training.rps.app.App();
```
a moze inna nazwa niz App ? wtedy ladniej by to wygladalo :D poza tym nie mam pojecia co robi to App, nazwa jest dosc malo mowi :)

PrzemekSwiderski replied:
Zmienione :)

aPusio commented on src/main/java/sda/training/HibernateFactory.java:
```diff
@@ -4,19 +4,25 @@
         configuration.setProperty("hibernate.connection.driver_class", "com.mysql.cj.jdbc.Driver");
         configuration.setProperty("hibernate.hbm2ddl.auto", "update");
 
-        configuration.addAnnotatedClass(Car.class);
```
nie musiales usuwac Car zeby wszystko w learning ciagle dzialalo :)

aPusio commented on src/main/java/sda/training/rps/app/App.java:
```diff
@@ -32,21 +30,57 @@ private void runMenu() {
     private boolean getPlayersChoice() {
+        MainMenu option = MainMenu.getOptionByIndex(scanner.scannerInt());
 
+        switch (option) {
```
❤️

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
     }
 
     private Game createGame(Player player, int winStagesNo) {
+        game.setId(null);
```
to jest niepotrzebne hibernate i tak to nadpisze :P

PrzemekSwiderski replied:
nie do końca ale i tak zmienilem na 
`game = new Game();`

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
             addPoints(result);
-            continueGame = false;
+            System.out.println("Ty: " + game.getPlayerScore() + " \tPrzeciwnik: " + game.getComputerScore());
+            if (isContinuable(true)) {
```
skoro zawsze przekazujesz tutaj true to jaki jest sens przekazywania tego parametru :D?

PrzemekSwiderski replied:
Gdyż inaczej musiał bym przeciazac metode, gdyż w innym miejscy korzystam z 
`isContinuable(continueGame)`
a metoda wymaga argumentu boolean

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
+        System.out.print("Czy chcesz kontynuować?(y/n): ");
+        String playerChoice = scanner.scannerLine();
+        switch (playerChoice.toLowerCase()) {
+            case "n":
```
idiotoodporne, podoba mi sie :D

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
 import java.util.List;
+import java.util.Objects;
 
-public class GameService implements IGameService {
```
ta klasa jest zdecydowanie za duza i na pewno nie ma jednej odpowiedzialnosci

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
     }
 
-
     private int getPlayerChoice(int size) {
```
np ta funkcje dosyc latwo wyrzucic do innej klasy :) w koncu to player choice a nie game choice

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
-    private void loadScores() {
-        game.setPlayerScore(stageDao.findAllByGameWinByPlayer(game).size());
-        game.setComputerScore(stageDao.findAllByGameWinByComputer(game).size());
+    private void loadScores(Game oldGame) {
```
a moze klasa GameLoader ? jest tu kilka metod load... ktore mogly by tam wyladowac :P

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
+        }
+    }
+
+    private String getTime(Game game) {
```
jeszcze nie wiem co tu sie dzieje ale na pewno jest jakis formater, funkcj util do odejmowania czasu albo cos innego co by Ci te wszystkie linijki zastapila :)

aPusio commented on src/main/java/sda/training/rps/app/GameService.java:
```diff
@@ -1,136 +1,317 @@
+    }
 
-        return 0;
+    private int controlListAllGames(int page, Player player) {
```
co ta funkcja robi :o ?

aPusio commented on src/main/java/sda/training/rps/app/StageService.java:
```diff
@@ -1,49 +1,141 @@
+    private final ScannerSingleton scanner = ScannerSingleton.getInstance();
+    private final PlayerService playerService;
+
+    public StageService(PlayerService playerService) {
```
jak nie dasz dao w konstruktorze to nie zrobisz testow :(
(gdyby to nie byl final moglbys teoretycznie wystawic setter do dao ale konstruktor + final jest o wiele lepszym pomyslem :)  )

aPusio commented on src/main/java/sda/training/rps/app/StageService.java:
```diff
@@ -1,49 +1,141 @@
+
+    private Result[][] getOutcome() {
+        return new Result[][]{
+                {Result.DRAW, Result.LOSE, Result.LOSE, Result.LOSE, Result.WIN, Result.WIN, Result.WIN},
```
chyba rozumiem pomysl, to jest taka mapa zaleznosci co z czym wygrywa tak ? 
troche malo to czytelne :( 

zaproponuje inny pomysl :) 
Mapa kto z kim wygrywa, jesli czegos tam nie ma to znaczy ze przegrywasz. 
to wydaje sie dosc naturalne map.get(scisors) powinienes dostac liste z czym wygrywaja nozyczki, latwo to przeszukac a i zapis byl by calkiem czytelny 

map.put(scisors, List.of(paper, salami, morkyHerbatnik)) <- od razu z kim wygrywa

aPusio commented on src/main/java/sda/training/rps/app/StageService.java:
```diff
@@ -1,49 +1,141 @@
+        if (stages.size() < 10) {
+            return Move.getMoveById(random.nextInt(7));
+        } else {
+            List<Move> moves = stages.stream().map(Stage::getPlayerMove).collect(Collectors.toList());
```
wnetrze tego else spokojnie mozesz podzielic na kilka funkcji, bedze czytelniej 

tak wlasciwie sam wprowadziles podzial robiac dodatkowe entery

aPusio commented on src/main/java/sda/training/rps/dao/GameDao.java:
```diff
@@ -1,39 +1,114 @@
     public Game findById(int id) {
-        return null;
+
+        try (SessionFactory factory = new HibernateFactory().getSessionFactory();
```
robiac tutaj new HibernateFactory zamykasz sie na podmiane tego na np testowe hibernateFactory, 
lepszym pomyslem bedzie pole HibernateFactory + konstruktor do tego :)

aPusio replied:
ja bym wolal chyba nawet w glownym mainie nawet otworzyc sessionFacory a potem tylko otwierac sesje w kazdym dao :)

aPusio commented on src/main/java/sda/training/rps/dao/GameDao.java:
```diff
@@ -1,39 +1,114 @@
+                    .getResultList();
+            session.close();
+            factory.close();
+            return Objects.requireNonNullElse(games, Collections.emptyList());
```
o to jest nice ! :)

aPusio commented on src/main/java/sda/training/rps/dao/PlayerDao.java:
```diff
@@ -1,33 +1,65 @@
+                    .uniqueResultOptional();
+            session.close();
+            factory.close();
+            return optional.orElse(null);
```
to mzoe optional zamiast nulla :D ?

aPusio commented on src/main/java/sda/training/sos/game/Position.java:
```diff
@@ -0,0 +1,25 @@
+package sda.training.sos.game;
+
+public class Position {
```
o wciagnales tez i moje zmiany :D

aPusio replied:
ale to zniknie jak ja zrobie merge (o ile dostane approvale :) )

aPusio commented on pom.xml:
```diff
@@ -1,101 +1,110 @@
+        <dependency>
+            <groupId>org.hibernate</groupId>
+            <artifactId>hibernate-core</artifactId>
+            <version>5.6.2.Final</version>
```
zmiana formatowania to zawsze spory problem, bo potem wszystko sie konfliktuje :P nie jest to zbyt wielki problem ale radze z tym uwazac :D

andramil committed:
Revert "learning"

This reverts commit 0747d4278530bbab2797714ca0e7feca8ea7e4ef.

andramil committed:
Running With Scissors
Poprawa komentarzy:
- rename klasy APP
- podzielenie logiki wyboru ruchu komputera na mniejsze metody
- ogarniecie opcjonala
- uzpełnienie encji Car
